### PR TITLE
show chunks seperations in database, proper error handling and prettier config

### PR DESCRIPTION
**Pull request by ThakerKush**
### Summary

This pull request introduces a new feature for storing and retrieving chunks of text from a vector database, along with improved error handling for processing large documents.

### Details

- **New Feature: Chunk ID's in Database:**
  - Added functionality to store chunk ID's in the database, allowing for efficient retrieval of specific chunks of text.
  - Implemented a new `batchCreateChunksAndEmbeddings` function in `helper.ts` to handle the creation and storage of chunks.
  - Modified the `index.ts` file to return the chunked input, which includes the chunk ID's and corresponding text.

- **Improved Error Handling:**
  - Enhanced error handling in `web/app/actions/doers.ts` to provide more informative error messages when processing large documents.
  - Added a check in `index.ts` to prevent the processing of documents with more than 20 chunks.

- **Refactoring:**
  - Introduced a `prettier` configuration file to enforce consistent code formatting.
  - Removed unnecessary imports and code in `web/app/actions/doers.ts`.
  - Updated the `createMemory` function in `web/app/actions/doers.ts` to handle the chunked input from the backend.

- **Other Changes:**
  - Updated the `systemPrompt` and `template` variables in `prompts/prompt1.ts` (likely related to the new chunking feature).

> ✨ Generated with love by Kaizen ❤️

<details>
<summary>Original Description</summary>
feat: improve error handling for long documents
feat: add chunk id's to db
</details>

### CODE REVIEW

File: apps/cf-ai-backend/src/index.ts [hunk 1]
```diff
@@ -65,14 +64,18 @@ app.post("/api/add", zValidator("json", vectorObj), async (c) => {
   const { store } = await initQuery(c);
 
   console.log(body.spaces);
-  await batchCreateChunksAndEmbeddings({
+  const chunks = chunkText(body.pageContent, 1536);
+  if (chunks.length > 20) {
+    return c.json({ status: "error", message: "no chunks over 20" });
```
Comment (Dhravya) [inline apps/cf-ai-backend/src/index.ts]: I assume this message is shown to the user, maybe we can keep the errors consistent for the user and just say "This document is too large" or something?

File: apps/web/app/actions/doers.ts [hunk 2]
```diff
@@ -265,6 +262,40 @@ export const createMemory = async (input: {
...
+    status: string;
+    chunkedInput: string;
+    message?: string;
+  };
+
+  try {
+    if (response.status !== "ok") {
+      if (
+        response.status === "error" &&
+        response.message === "no chunks over 20"
```
Comment (Dhravya) [inline apps/web/app/actions/doers.ts]: matching the exact message can make it difficult to maintain, we can either:
1. use `data` as an numeric error code
2. prefix the message with 'ERROR:' and show the exact same error message to the user

File: apps/web/app/actions/doers.ts [hunk 3]
```diff
@@ -349,32 +380,10 @@ export const createMemory = async (input: {
     );
   }
 
-  try {
```
Comment (Dhravya) [inline apps/web/app/actions/doers.ts:352]: not sure if removing this is a good idea, this existed to make sure that there's 'consistency' between the vector store and our document store. are we still making sure that the data is consistent? if yes, yeah removing it is a great idea

**Review by Dhravya — CHANGES_REQUESTED** [review]

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "tabWidth": 2,
+  "trailingComma": "all",
+  "useTabs": false
```
Comment (Dhravya) [inline .prettierrc:4]: `useTabs` should be true, something I learnt recently

**Review by Dhravya — APPROVED** [review]

File: .prettierrc (hunk 4 above)
Comment (Dhravya) [reply .prettierrc:4]: .

**Review by Dhravya — CHANGES_REQUESTED** [review]

**Comment by Dhravya** [comment]
!feedback automatically changed title was not descriptive

File: apps/web/app/actions/doers.ts (hunk 3 above)
Comment (ThakerKush) [reply apps/web/app/actions/doers.ts:352]: This had to be moved to account for error response form the /api/add endpoint ( the error related to the maximum chunk size ) and yeah the vector store and d1 store should be consistent 

But there is an edge case here and in main, if one fails and the other one succeeds then they won't be consistent........right?

File: apps/web/app/actions/doers.ts (hunk 2 above)
Comment (ThakerKush) [reply apps/web/app/actions/doers.ts]: Yeah, gonna change this

File: apps/cf-ai-backend/src/index.ts (hunk 1 above)
Comment (ThakerKush) [reply apps/cf-ai-backend/src/index.ts]: How about I change this one to "We are unable to process documents this size just yet, try something smaller"?

File: apps/web/app/actions/doers.ts (hunk 3 above)
Comment (Dhravya) [reply apps/web/app/actions/doers.ts:352]: Yeah, we gotta make sure that doesn't happen. If it does maybe we should call the delete endpoints to get rid of them?

Comment (ThakerKush) [reply apps/web/app/actions/doers.ts:352]: Yes that or like a exponential back off retry for the failed db call, if that fails to then call the delete endpoint?

File: apps/cf-ai-backend/src/index.ts (hunk 1 above)
Comment (Dhravya) [reply apps/cf-ai-backend/src/index.ts]: yes works!

File: apps/web/app/actions/doers.ts (hunk 3 above)
Comment (Dhravya) [reply apps/web/app/actions/doers.ts:352]: Yes that would be absolutely amazing

Comment (Dhravya) [reply apps/web/app/actions/doers.ts:352]: we don't _need_ retries or backoffs, it's a good to have, as long as we just ensure both are the same

**Review by Dhravya — APPROVED** [review]

**Pull request merged**